front/src/common/RollingStockSelector/RollingStockModal.tsx

```diff
@@ -1,35 +1,78 @@
-import React, { useState, useEffect, useContext, useMemo } from 'react';
-import PropTypes from 'prop-types';
-import Loader from 'common/Loader';
+import React, { useState, useEffect, useContext, useMemo, MutableRefObject } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
 import { setFailure } from 'reducers/main';
 import { useTranslation } from 'react-i18next';
-import CheckboxRadioSNCF from 'common/BootstrapSNCF/CheckboxRadioSNCF';
 import { BsLightningFill } from 'react-icons/bs';
 import { MdLocalGasStation } from 'react-icons/md';
-import './RollingStock.scss';
+import { isEmpty, sortBy } from 'lodash';
+
+import { LightRollingStock, osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { RootState } from 'reducers';
+import { getRollingStockID } from 'reducers/osrdconf/selectors';
+import Loader from 'common/Loader';
+import CheckboxRadioSNCF from 'common/BootstrapSNCF/CheckboxRadioSNCF';
 import InputSNCF from 'common/BootstrapSNCF/InputSNCF';
 import ModalBodySNCF from 'common/BootstrapSNCF/ModalSNCF/ModalBodySNCF';
-import { getRollingStockID } from 'reducers/osrdconf/selectors';
 import { ModalContext } from 'common/BootstrapSNCF/ModalSNCF/ModalProvider';
-import { isEmpty, some, sortBy } from 'lodash';
-import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
 import RollingStockEmpty from './RollingStockEmpty';
 import RollingStockCard from './RollingStockCard';
+import './RollingStock.scss';
 
-function RollingStockModal(props) {
-  const { ref2scroll } = props;
+interface Filters {
+  text: string;
+  elec: boolean;
+  thermal: boolean;
+}
+function rollingStockPassesSearchedStringFilter(
+  name: string,
+  metadata: LightRollingStock['metadata'],
+  filters: Filters
```

Comment:
```suggestion
  filters: Filters
```
I think you can either pass the whole Filter to your 2 functions (rollingStockPassesSearchedStringFilter and rollingStockPassesEnergeticModeFilters) or only the needed fields (text for the first one, filterElec and filterThermal for the 2nd one)

It's a detail though :)

Reply:
great observation :) stringFilter does the first version and energeticFilter does the second.

Reply:
fixed